databricks_cli/workspace/types.py

```diff
@@ -0,0 +1,97 @@
+# Databricks CLI
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from click import ParamType
+
+
+class WorkspaceLanguage(object):
+    SCALA = 'SCALA'
+    PYTHON = 'PYTHON'
+    SQL = 'SQL'
+    R = 'R'
+    ALL = [SCALA, PYTHON, SQL, R]
+    EXTENSIONS = ['.scala', '.py', '.sql', '.r', '.R']
+
+    @classmethod
+    def to_language(cls, path):
+        ext = cls.get_extension(path)
+        if ext == '.scala':
+            return cls.SCALA
+        elif ext == '.py':
+            return cls.PYTHON
+        elif ext == '.sql':
+            return cls.SQL
+        elif ext == '.r':
+            return cls.R
+        elif ext == '.R':
```

Comment:
Why is R special?

Reply:
Can you comment on this? Why do we only support upper case for R? I feel it is better to keep consistent.

Reply:
This was brought up in the design doc by @mengxr. I don't have a strong preference either way.

Reply:
We can make all of them case insensitive.